src/main/java/seedu/duke/models/schema/Student.java

```diff
@@ -120,11 +119,21 @@ public void addModuleSchedule(String moduleCode, int targetSem) throws InvalidOb
      * @param moduleCode The code of the module to be completed.
      */
     public void completeModuleSchedule(String moduleCode) throws InvalidObjectException {
+        //check prereq
+        if (schedule.canCompleteModule(moduleCode)) {
```

Comment:
For functions we can try structuring it using happy path principle? So
```
void function(){
   if(fail){
       return;
   }
   happy path;
 
}
```
I think it'll help with readability, thoughts?